### PR TITLE
DDF-1262 added CQL query functionality to cache:search command

### PR DESCRIPTION
**Pull request by wmcnalli**
@pklinef 
@millerw8 
@jlcsmith 
@kcwire 
@beyelerb 
@emanns95
@jrnorth 
@kguerriero

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-catalog/127)

### CODE REVIEW

**Comment by wmcnalli** [comment]
@kguerriero can you hero this?

**Comment by kcwire** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

<sup>**[core/catalog-core-standardframework/pom.xml, line 94 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/127#-JuqXA6ewN8fQMkGGJUR)** ([raw file](https://github.com/codice/ddf-catalog/blob/eab5497c08ed8b7fb3c92d4beca4544dcfd91f1a/core/catalog-core-standardframework/pom.xml#L94)):</sup>
do we not have geotools in dependency management somewhere?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by wmcnalli** [comment]
Review status: 6 of 9 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

<sup>**[core/catalog-core-standardframework/pom.xml, line 94 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/127#-JuqXA6ewN8fQMkGGJUR)** ([raw file](https://github.com/codice/ddf-catalog/blob/eab5497c08ed8b7fb3c92d4beca4544dcfd91f1a/core/catalog-core-standardframework/pom.xml#L94)):</sup>
We do... fixed in the other poms where i found it but this dependency is no longer needed.  Gone!

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by kguerriero** [comment]
Review status: 6 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 66 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/127#-Juq_glo6aI51XyZN2EE)** ([raw file](https://github.com/codice/ddf-catalog/blob/eab5497c08ed8b7fb3c92d4beca4544dcfd91f1a/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L66)):</sup>
Will searchPhrase be set by felix somewhere? It seems that it will always be null at this point.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by wmcnalli** [comment]
Review status: 6 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

<sup>**[core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java, line 66 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-catalog/127#-Juq_glo6aI51XyZN2EE)** ([raw file](https://github.com/codice/ddf-catalog/blob/eab5497c08ed8b7fb3c92d4beca4544dcfd91f1a/core/catalog-core-cachecommands/src/main/java/org/codice/ddf/commands/cache/SearchCommand.java#L66)):</sup>
It gets picked up as an argument to the command - see the `@Argument` annotation above.  If I do `cache:search abc`, searchPhrase will be set to `"abc"`; if I don't pass in an argument, we do a wildcard search.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by kguerriero** [comment]
Reviewed 2 of 2 files at r2.
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by kguerriero** [comment]
Builds and all tests pass. Verified that cql search works as expected

---

Reviewed 6 of 7 files at r1, 2 of 2 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by jrnorth** [comment]
LGTM

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by kguerriero** [comment]
Successfully built and ran all tests. Verified that cql performs as expected. 

---

Review status: 6 of 8 files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/127)

**Comment by kcwire** [comment]
@kguerriero if all is good I will merge this today.

File: core/catalog-core-cachecommands/pom.xml [hunk 1]
```diff
@@ -78,7 +78,6 @@
         <dependency>
             <groupId>org.geotools</groupId>
             <artifactId>gt-cql</artifactId>
```
Comment (pklinef) [inline core/catalog-core-cachecommands/pom.xml]: There are bugs with gt-cql and how it handles zulu time and time zones.  If you can, you should use version 13 or later to get those fixes.

Comment (kcwire) [inline core/catalog-core-cachecommands/pom.xml]: @wmcnalli Can you up the gt-cql version number?

Comment (wmcnalli) [inline core/catalog-core-cachecommands/pom.xml]: Assume we just want to up it here and leave the managed version alone?  New ticket to up the version globally?

**Pull request merged**